oci.go

```diff
@@ -129,16 +129,66 @@ func stopContainer(podStatus vc.PodStatus) error {
 }
 
 // processCgroupsPath process the cgroups path as expected from the
-// OCI runtime specification. It returns a complete path to the path
-// that should be created and used.
-func processCgroupsPath(ociSpec specs.Spec) (string, error) {
+// OCI runtime specification. It returns a list of complete paths
+// that should be created and used for every specified resource.
+func processCgroupsPath(ociSpec specs.Spec) ([]string, error) {
+	var cgroupsPathList []string
+
 	if ociSpec.Linux.CgroupsPath == "" {
-		return "", nil
+		return []string{}, nil
+	}
+
+	if ociSpec.Linux.Resources == nil {
+		return []string{}, nil
+	}
+
+	if ociSpec.Linux.Resources.Memory != nil {
+		memCgroupsPath, err := processCgroupsPathForResource(ociSpec, "memory")
+		if err != nil {
+			return []string{}, err
+		}
+
+		cgroupsPathList = append(cgroupsPathList, memCgroupsPath)
+	}
+
+	if ociSpec.Linux.Resources.CPU != nil {
+		cpuCgroupsPath, err := processCgroupsPathForResource(ociSpec, "cpu")
+		if err != nil {
+			return []string{}, err
+		}
+
+		cgroupsPathList = append(cgroupsPathList, cpuCgroupsPath)
+	}
+
+	if ociSpec.Linux.Resources.Pids != nil {
+		pidsCgroupsPath, err := processCgroupsPathForResource(ociSpec, "pids")
+		if err != nil {
+			return []string{}, err
+		}
+
+		cgroupsPathList = append(cgroupsPathList, pidsCgroupsPath)
+	}
+
+	if ociSpec.Linux.Resources.BlockIO != nil {
+		blkIOCgroupsPath, err := processCgroupsPathForResource(ociSpec, "blkio")
+		if err != nil {
+			return []string{}, err
+		}
+
+		cgroupsPathList = append(cgroupsPathList, blkIOCgroupsPath)
+	}
+
+	return cgroupsPathList, nil
+}
+
+func processCgroupsPathForResource(ociSpec specs.Spec, resource string) (string, error) {
+	if resource == "" {
+		return "", fmt.Errorf("resource is empty")
```

Comment:
All these could be replaced by errors.New(), that avoids Errorf scanning the format string looking of % markers, not that we really care about speed in error paths...